fe/src/main/java/org/apache/doris/catalog/Index.java

```diff
@@ -0,0 +1,180 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.catalog;
+
+import org.apache.doris.analysis.IndexDef;
+import org.apache.doris.common.io.Text;
+import org.apache.doris.common.io.Writable;
+import org.apache.doris.thrift.TIndexType;
+import org.apache.doris.thrift.TOlapTableIndex;
+
+import com.google.common.base.Strings;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Internal representation of index, including index type, name, columns and comments.
+ * This class will used in olaptable
+ */
+public class Index implements Writable {
+    private String indexName;
+    private List<String> columns;
+    private IndexDef.IndexType type;
+    private String comment;
+
+    public Index(String indexName, List<String> columns, IndexDef.IndexType type, String comment) {
+        this.indexName = indexName;
+        this.columns = columns;
+        this.type = type;
+        this.comment = comment;
+    }
+
+    public Index() {
+        this.indexName = null;
+        this.columns = null;
+        this.type = null;
+        this.comment = null;
+    }
+
+    public String getIndexName() {
+        return indexName;
+    }
+
+    public void setIndexName(String indexName) {
+        this.indexName = indexName;
+    }
+
+    public List<String> getColumns() {
+        return columns;
+    }
+
+    public void setColumns(List<String> columns) {
+        this.columns = columns;
+    }
+
+    public IndexDef.IndexType getType() {
+        return type;
+    }
+
+    public void setType(IndexDef.IndexType type) {
+        this.type = type;
+    }
+
+    public String getComment() {
+        return comment;
+    }
+
+    public void setComment(String comment) {
+        this.comment = comment;
+    }
+
+    public JSONObject toJson() {
```

Comment:
We can use annotation to finish it in a simple way, you can refer to Tag related class.